core/database/arango_unix_client.py

```
@@ -0,0 +1,253 @@
+
+logger = logging.getLogger(__name__)
+
+
+class ArangoUnixClient:
+    """
+    ArangoDB client optimized for Unix socket connections.
+
+    Falls back to HTTP gracefully if Unix socket is not available.
+    """
+
+    DEFAULT_UNIX_SOCKET = "/tmp/arangodb.sock"
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/arangodb.sock"

(S108)